### PR TITLE
Update quickstart doc

### PR DESCRIPTION
**Pull request by smitterl**
1. Minor update aligning references to official documentation and sections.
2. For users not familiar with pip give a hint how to call `avocado` after
   running `pip install --user ...`

Signed-off-by: Sebastian Mitterle <smitterl@redhat.com>

### CODE REVIEW

File: docs/source/quickstart/index.rst [hunk 1]
```diff
@@ -184,14 +183,14 @@ It is super easy, just run the follow command::
...
+.. note:: For more methods, please visit the Installing section on
+          Avocado User's Guide.
 
 Documentation
 =============
 
-Please see Contents for full documentation, including installation methods,
-tutorials and API.
+Please see the `latest documentation <https://avocado-framework.readthedocs.io/en/latest/index.html>`_
+for full documentation, including installation methods, tutorials and API.
```
Comment (beraldoleal) [inline docs/source/quickstart/index.rst]: The only "problem" that I can see here, is that we use this `quickstart/index.rst` on both: README and readthedocs index page.

So if I'm visiting the readthedocs page and click on this link I will be redirected to the same page, right?

Comment (clebergnu) [reply docs/source/quickstart/index.rst]: One other side effect, which I don't know if it's intended, is that that's a difference between the "latest" version (tracks master) and the latest *released* version (which I updated after each release).

**Comment by beraldoleal** [comment]
@smitterl also looks like the tests are failing because "Signed-off-by:" is missing:

https://avocado-framework.readthedocs.io/en/latest/guides/contributor/chapters/styleguides.html

File: docs/source/quickstart/index.rst (hunk 1 above)
Comment (smitterl) [reply docs/source/quickstart/index.rst]: @beraldoleal , @clebergnu Thank you! I agree. There seem to be three scenarios:

1. User is on github (README, or docs/source/quickstart/index.rst)
2. User is on latest documentation (in development)
3. User is on some release's documentation (who doesn't want to click to another version's documentation)

I think we can serve all by

a. linking to both latest and last release's documentation
b. adding "or browse this site for more content."

quickstart continues to serve both as README and Welcome page.

I updated the PR accordingly. Please, let me know what you think.

**Comment by smitterl** [comment]
I added additionally a short note about where pip --user installs commands for people not too familiar with Python.

File: docs/source/guides/user/chapters/installing.rst [hunk 2]
```diff
@@ -26,7 +26,8 @@ with a single command::
 
 This will fetch the Avocado package (and possibly some of its dependecies) from
 the PyPI repository, and will attempt to install it in the user's home
-directory (usually under ``~/.local``).
+directory (usually under ``~/.local``), which you might want to add to your
```
Comment (clebergnu) [inline docs/source/guides/user/chapters/installing.rst]: Nice.

File: docs/source/quickstart/index.rst [hunk 3]
```diff
@@ -30,10 +30,10 @@ You probably noticed that we used ``/bin/true`` as a test, and in accordance
 with our expectations, it passed! These are known as `simple tests`, but there
 is also another type of test, which we call `instrumented tests`.
 
-.. tip:: See more at the "Test types" sections to understand Avocado's test types.
+.. tip:: See more at the "Test types" section on the Avocado User's Guide.
 
-Why should I use?
-=================
+Why should I use it?
```
Comment (clebergnu) [inline docs/source/quickstart/index.rst]: Now that you've fixed it, I'm asking myself how I was reading it without catching it.

**Pull request merged**